src/features/import/components/ImportFromIMDB.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react';
+import { CSVReader } from 'react-papaparse';
+import { camelCase } from 'lodash';
+import { useDispatch, useSelector } from 'react-redux';
+import { unwrapResult } from '@reduxjs/toolkit';
+import { toast, ToastContainer } from 'react-toastify';
+
+import { Button } from 'app/components/Button';
+
+import { selectBookmarksIndex } from 'features/bookmarks/selectors';
+import {
+  addEmptyBookmarksDocToIndex,
+  addManyBookmarks,
+} from 'features/bookmarks/asyncThunks';
+import { selectRatingsIndex } from 'features/ratings/selectors';
+import {
+  addIndexKeyToRatingsIndex,
+  addManyRatings,
+} from 'features/ratings/asyncThunks';
+import { isIMDBRatingValid } from 'features/import/utils';
+import { RatingsImportSource } from 'app/constants';
+
+export function ImportFromIMDB(): JSX.Element {
+  const dispatch = useDispatch();
+  const [isImporting, setIsImporting] = useState(false);
+  const bookmarksIndex = useSelector(selectBookmarksIndex);
+  const ratingsIndex = useSelector(selectRatingsIndex);
+  const [validIMDBRatings, setValidIMDBRatings] = useState<any[]>([]);
+
+  const handleOnDrop = (
+    ratings: {
+      data: any;
+      errors: any[];
+    }[]
+  ) => {
+    const validRatings = ratings
+      .filter(
+        (rating) => rating.errors.length === 0 && isIMDBRatingValid(rating.data)
+      )
+      .map((validRating) => validRating.data);
+    setValidIMDBRatings(validRatings);
+  };
```

Comment:
I got this error, as one of my movies' director fields was empty. Upload worked after I filled the field with text.

```
{
  "name": "TypeError",
  "message": "Cannot read property 'toString' of undefined",
  "stack": "TypeError: Cannot read property 'toString' of undefined\n    at DocCache.get (http://localhost:3000/static/js/0.chunk.js:16588:42)\n    at CeramicClient.loadDocument (http://localhost:3000/static/js/0.chunk.js:18514:30)\n    at Module.<anonymous> (http://localhost:3000/static/js/0.chunk.js:159255:44)\n    at Generator.next (<anonymous>)\n    at http://localhost:3000/static/js/0.chunk.js:159141:67\n    at new Promise (<anonymous>)\n    at push../node_modules/kontext-common/lib/esm/apis/bookmarks.js.__awaiter (http://localhost:3000/static/js/0.chunk.js:159120:10)\n    at Module.addManyBookmarkDocsToBookmarksDoc (http://localhost:3000/static/js/0.chunk.js:159250:10)\n    at addManyBookmarkDocsToBookmarksDoc (http://localhost:3000/static/js/main.chunk.js:421:72)\n    at http://localhost:3000/static/js/main.chunk.js:1787:127\n    at async handleClickImport (http://localhost:3000/static/js/main.chunk.js:3406:44)"
}
```

I guess the solution is here. Mapping any `undefined` to `""`.

Reply:
Good catch! Was actually a race condition in the actions but should be fixed now